pkg/api/install/install.go

```diff
@@ -19,4 +22,458 @@ import (
 	_ "github.com/openshift/origin/pkg/security/api/install"
 	_ "github.com/openshift/origin/pkg/template/api/install"
 	_ "github.com/openshift/origin/pkg/user/api/install"
+
+	kv1 "k8s.io/kubernetes/pkg/api/v1"
+
+	watchapi "k8s.io/kubernetes/pkg/watch"
+	watchv1 "k8s.io/kubernetes/pkg/watch/versioned"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	oauthapi "github.com/openshift/origin/pkg/oauth/api"
+	projectapi "github.com/openshift/origin/pkg/project/api"
+	routeapi "github.com/openshift/origin/pkg/route/api"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	userapi "github.com/openshift/origin/pkg/user/api"
+
+	authorizationv1 "github.com/openshift/origin/pkg/authorization/api/v1"
+	buildv1 "github.com/openshift/origin/pkg/build/api/v1"
+	deployv1 "github.com/openshift/origin/pkg/deploy/api/v1"
+	imagev1 "github.com/openshift/origin/pkg/image/api/v1"
+	oauthv1 "github.com/openshift/origin/pkg/oauth/api/v1"
+	projectv1 "github.com/openshift/origin/pkg/project/api/v1"
+	routev1 "github.com/openshift/origin/pkg/route/api/v1"
+	templatev1 "github.com/openshift/origin/pkg/template/api/v1"
+	userv1 "github.com/openshift/origin/pkg/user/api/v1"
 )
+
+func init() {
+	// This is a "fast-path" that avoids reflection for common types. It focuses on the objects that are
+	// converted the most in the cluster.
+	// TODO: generate one of these for every external API group - this is to prove the impact
```

Comment:
Would be in Kube, will create.

On Thu, Jun 16, 2016 at 10:32 AM, Michal Fojtik <notifications@github.com>
wrote:

> In pkg/api/install/install.go
> <https://github.com/openshift/origin/pull/9365#discussion_r67355316>:
>
> >  )
> > +
> > +func init() {
> > +	// This is a "fast-path" that avoids reflection for common types. It focuses on the objects that are
> > +	// converted the most in the cluster.
> > +	// TODO: generate one of these for every external API group - this is to prove the impact
>
> follow up issue?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9365/files/0768a8081118df921fba9b5cf1edfca0bac78367#r67355316>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p5jGZNqS8uH9-AE30Cfz6Y71iz_bks5qMV5zgaJpZM4I27dv>
> .
>